lib/OpenQA/WebAPI/Controller/Test.pm

```diff
@@ -570,30 +570,47 @@ sub prepare_job_results {
     my %descriptions = map { $_->name => $_->description } @descriptions;
 
     my @wanted_jobs = grep {
-            (not $states         or $states->{$_->state})
-        and (not $results        or $results->{$_->result})
-        and (not $archs          or $archs->{$_->ARCH})
-        and (not $machines       or $machines->{$_->MACHINE})
-        and (not $failed_modules or $_->result eq OpenQA::Jobs::Constants::FAILED)
+              (not $states         or $states->{$_->state})
+          and (not $results        or $results->{$_->result})
+          and (not $archs          or $archs->{$_->ARCH})
+          and (not $machines       or $machines->{$_->MACHINE})
+          and (not $failed_modules or $_->result eq OpenQA::Jobs::Constants::FAILED)
     } @$jobs;
-    my @jobids = map { $_->id } @wanted_jobs;
+    my @jobids                = map { $_->id } @wanted_jobs;
     my $failed_modules_by_job = $schema->resultset('JobModules')->search(
-        {job_id => {-in => [@jobids]}, result => 'failed'},
-        {select   => [qw(name job_id)], order_by => 't_updated'},
+        {job_id => {-in => [@jobids]}, result   => 'failed'},
+        {select => [qw(name job_id)],  order_by => 't_updated'},
     );
     my %failed_modules_by_job;
     for my $item ($failed_modules_by_job->all) {
         push @{$failed_modules_by_job{$item->job_id}}, $item->name;
     }
+    my %children_by_job;
+    my %parents_by_job;
+    my $s = $schema->resultset('JobDependencies')->search(
+        {
+            -or => [
+                parent_job_id => {-in => \@jobids},
+                child_job_id  => {-in => \@jobids},
+            ],
+        });
+    while (my $dep = $s->next) {
+        my $pid = $dep->parent_job_id;
+        my $kid = $dep->child_job_id;
```

Comment:
These variables are only used once and the lines below are still quite short. I guess I'd avoid the extra lines.

Reply:
They are used twice, not once :)

I can still change it, though

Reply:
Somehow I don't see where.

Reply:
I'm not seeing it either 🤔

Reply:
btw awesome name, `$kid`

Reply:
I don't see it either

Reply:
Oh sorry, I didn't look at the next lines and remembered it wrongly that each variable was used in both lines.
Will change